lib/rspec/mocks/error_generator.rb

```diff
@@ -268,6 +268,17 @@ def unexpected_arguments_message(expected_args_string, actual_args_string)
       def error_message(expectation, args_for_multiple_calls)
         expected_args = format_args(expectation.expected_args)
         actual_args = format_received_args(args_for_multiple_calls)
+
+        if RUBY_VERSION >= "3" && RSpec::Support::RubyFeatures.kw_args_supported? && expected_args == actual_args
+          expected_hash = expectation.expected_args.last
+          actual_hash = args_for_multiple_calls.last.last
+          if Hash === expected_hash && Hash === actual_hash &&
+            (Hash.ruby2_keywords_hash?(expected_hash) ^ Hash.ruby2_keywords_hash?(actual_hash))
```

Comment:
Is `^` same as `!=`? I find `^` a bit cryptic.

Reply:
Ah good point 😓

Reply:
Fixed.